### PR TITLE
Profile admin

### PR DESCRIPTION
**Pull request by namila007**
User profile view aded issue #18

### CODE REVIEW

File: src/pages/Profile/Profile.vue [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
 
 <script>
+import UserCard from './UserCard.vue'
+import ProfileView from './ProfileView.vue'
 export default {
-  name: 'Profile'
+  name: 'Profile',
+  data () {
+    return {
+      User: null
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: User should be simple. follow `camelCase`

File: src/pages/Profile/Profile.vue [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
+      </v-layout>
+        </v-container>
+        </div>
+    </v-flex>
+
+    
 </template>
 
 <script>
+import UserCard from './UserCard.vue'
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: No need for this, just output all the values like in the `src/pages/Users/User.vue`
You can copy paste it also

File: src/pages/Profile/ProfileView.vue [hunk 3]
```diff
@@ -0,0 +1,132 @@
+<template>
```
Comment (kasvith) [inline src/pages/Profile/ProfileView.vue]: All editings should be done in the settings page. remove this one

File: src/pages/Profile/UserCard.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<template>
```
Comment (kasvith) [inline src/pages/Profile/UserCard.vue]: you can do this in single file, please refer User.vue

**Review by kasvith — CHANGES_REQUESTED** [review]
Please remove all the redundant files and make this clean like User.vue in the src/pages/Users/User.vue

Also follow camelCase for code

**Comment by kasvith** [comment]
Prefer something like this 
image

File: src/pages/Profile/Profile.vue [hunk 5]
```diff
@@ -1,10 +1,43 @@
...
 export default {
-  name: 'Profile'
+  name: 'Profile',
+  data () {
+    return {
+      User: null
+    }
+  },
+  mounted () {
+    this.User = this.$store.state.user.user
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: You can use a getter here

this.$store.getters['user/user']

**Review by namila007 — COMMENTED** [review]
new changes

File: src/pages/Profile/Profile.vue [hunk 6]
```diff
@@ -1,13 +1,110 @@
...
 <script>
 export default {
-  name: 'Profile'
+  name: 'Profile',
+  data () {
+    return {
+      user: null
+    }
+  },
+  mounted () {
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: You can use a computed property, so you dont need to fetch data when mounted

More details are available at https://github.com/alentar/rpms-admin/blob/c8e980f41fa66785db32d16e9abc201888ed54d3/src/components/Application/Menus/UserMenu.vue#L55

see that i get directly user from the store and i can use him just like i use a variable in data
this is the preferred way for achieving this task

File: src/pages/Profile/Profile.vue [hunk 7]
```diff
@@ -1,13 +1,110 @@
 <template>
-  <div>Profile</div>
+ 
+    <v-flex class="white elevation-2 fluid">
+        <v-toolbar flat dense class="blue darken-3" dark>
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: Can you remove toolbar ?
It's for being similar as other pages

For settings we can group using this, but in here it seems unnecessary

File: src/pages/Profile/Profile.vue [hunk 8]
```diff
@@ -1,13 +1,110 @@
...
+    </v-flex>    
 </template>
 
 <script>
 export default {
-  name: 'Profile'
+  name: 'Profile',
+  data () {
+    return {
+      user: null
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: Check the other comments, you can use computed property directly [Computed Property](https://vuejs.org/v2/guide/computed.html)

**Review by kasvith — COMMENTED** [review]
Please fix follows and then we will be good to go

File: src/pages/Profile/Profile.vue (hunk 6 above)
Comment (kasvith) [reply src/pages/Profile/Profile.vue]: this change will make your code much more simplified

File: src/pages/Profile/Profile.vue [hunk 9]
```diff
@@ -1,13 +1,110 @@
...
+                </v-layout>
+                <br>
+                <v-layout row md12 wrap>
+                  <v-flex md5 xs12 column>
+                  <p class="text"> NIC</p>
+                    {{user.nic}}
+                  </v-flex>
+                  <v-flex md6 offset-md1 column>
+                    <p class="text">Gender</p>
+                    {{user.gender}}
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: There is a vue filter for capitalize first letter of any string
`{{  user.gender | capitalizefirstletter }}`

will capitalize first letter of any sentence or word, use as you wish i see many simple case strings

File: src/pages/Profile/Profile.vue [hunk 10]
```diff
@@ -1,13 +1,110 @@
...
+                <div>
+                <v-avatar
+                :tile="false"
+                :size="150"
+                class="grey lighten-4"
+                align-center
+                >
+                <img src="/static/user.png" alt="avatar">    
+                </v-avatar>   
+                <h4 class="title">{{user.name}}
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: add spaces between brackets in every place 

`{{  user.name }}` is prefered

Comment (kasvith) [reply src/pages/Profile/Profile.vue]: I think outputting name here is not necessary, we already have it in other side....

but please follow the rule, `{{ variable }}`

File: src/pages/Profile/Profile.vue [hunk 11]
```diff
@@ -1,13 +1,110 @@
...
+        <v-toolbar-title>User Profile</v-toolbar-title>
+        </v-toolbar>
+        <div class="pl-4 pr-4 pt-3 pb-2">
+          <v-container grid >
+            <v-layout row wrap>
+            <v-flex md3 xs12 sm6>
+              <v-layout wrap>
+                <div>
+                <v-avatar
+                :tile="false"
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: set this true for get squared profile picture

File: src/pages/Profile/Profile.vue [hunk 12]
```diff
@@ -1,13 +1,110 @@
...
+                  <v-flex md6 offset-md1 column>
+                    <p class="text">Role</p>
+                    {{user.role}}
+                  </v-flex>
+                </v-layout>
+                <br>
+                <v-layout row wrap>
+                  <v-flex>
+                    <p class="text">Contacts</p>
+                    <v-list-tile v-for="contact in user.contacts" :key="contact">
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: also add `v-if="user.contacts.length > 0"` for hide this when no contacts are there

**Review by kasvith — COMMENTED** [review]
Fix follows

File: src/pages/Profile/Profile.vue (hunk 11 above)
Comment (namila007) [reply src/pages/Profile/Profile.vue]: circle is best for this design

File: src/pages/Profile/Profile.vue (hunk 10 above)
Comment (namila007) [reply src/pages/Profile/Profile.vue]: done

File: src/pages/Profile/Profile.vue (hunk 9 above)
Comment (namila007) [reply src/pages/Profile/Profile.vue]: done

File: src/pages/Profile/Profile.vue (hunk 8 above)
Comment (namila007) [reply src/pages/Profile/Profile.vue]: cant change everything , so mount is good for this simple form

File: src/pages/Profile/Profile.vue (hunk 6 above)
Comment (namila007) [reply src/pages/Profile/Profile.vue]: done

**Comment by namila007** [comment]
here is the final one.

File: src/pages/Profile/Profile.vue [hunk 13]
```diff
@@ -1,13 +1,107 @@
...
 export default {
-  name: 'Profile'
+  name: 'Profile',
+  computed: {
+    user () {
+      return this.$store.getters['user/user']
+    }
+  },
+  components: {
+    filters: {
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: you dont need this,

just use {{ variable | capitalizefirstletter }} anywhere in the code, its globally installed for project

Comment (kasvith) [reply src/pages/Profile/Profile.vue]: When ever you need to capitalize a string only first letter use {{ yourVariable | capitalizefirstletter }}

Comment (kasvith) [reply src/pages/Profile/Profile.vue]: just removing this will work, remove entire component part

**Review by kasvith — COMMENTED** [review]
Please remove this filter component

File: src/pages/Profile/Profile.vue (hunk 13 above)
Comment (kasvith) [reply src/pages/Profile/Profile.vue]: https://github.com/alentar/rpms-admin/blob/c8e980f41fa66785db32d16e9abc201888ed54d3/src/main.js#L32

It's already installed and configured

File: src/pages/Profile/Profile.vue [hunk 14]
```diff
@@ -82,13 +82,7 @@ export default {
     }
   },
   components: {
```
Comment (kasvith) [inline src/pages/Profile/Profile.vue]: remove this also, its redundant

Comment (kasvith) [reply src/pages/Profile/Profile.vue]: remove entire `component: { }`

**Review by kasvith — APPROVED** [review]

**Comment by kasvith** [comment]
Tnx @namila007

**Pull request merged**